src/backend/src/utils/reimbursement-requests.utils.ts

```
@@ -202,17 +202,70 @@ const createNewProducts = async (products: ReimbursementProductCreateArgs[], rei
   }
 };
 
+/**
+ * Validates that the given user is on the finance team.
+ *
+ * @param user The user to validate.
+ * @throws {AccessDeniedException} Fails validation when user is not on the
+ * finance team.
+ */
 export const validateUserIsPartOfFinanceTeam = async (user: User) => {
+  const isUserAuthorized = await isUserOnFinanceTeam(user);
+
+  if (!isUserAuthorized) {
+    throw new AccessDeniedException(`You are not a member of the finance team!`);
+  }
+};
+
+/**
+ * Determines if a user is part of the finance team.
+ *
+ * To be used for Prisma input validation of a plain User, as opposed to
+ * <code>isAuthUserOnFinance</code>, which uses the additional fields
+ * produced by authUserQueryArgs that are not in the User type by default.
+ *
+ * @param user the user to authenticate
+ * @returns whether the user is on the finance team
+ * @throws {HttpException} if finance team not found in database
+ */
+export const isUserOnFinanceTeam = async (user: User): Promise<boolean> => {
+  if (!process.env.FINANCE_TEAM_ID) {
+    console.warn('FINANCE_TEAM_ID not in env');
```

this should just throw an error if this env variable isn't set